File: compose-demo/src/main/kotlin/dtos.kt
/* Options:
Date: 2024-03-01 15:12:24
Version: 8.12
Tip: To override a DTO option, remove "//" prefix before updating
BaseUrl: https://localhost:5001

//Package: 
//AddServiceStackTypes: True
//AddResponseStatus: False
//AddImplicitVersion: 
//AddDescriptionAsComments: True
//IncludeTypes: 
//ExcludeTypes: 
//InitializeCollections: True
//TreatTypesAsStrings: 
//DefaultImports: java.math.*,java.util.*,net.servicestack.client.*
*/

import java.math.*
import java.util.*
import net.servicestack.client.*


@Route("/hello")
// @Route("/hello/{Name}")
open class Hello : IReturn<HelloResponse>, IGet
{
    var name:String? = null
    companion object { private val responseType = HelloResponse::class.java }
    override fun getResponseType(): Any? = Hello.responseType
}

open class SearchFiles : IReturn<SearchFilesResponse>, IGet
{
    var pattern:String? = null
    companion object { private val responseType = SearchFilesResponse::class.java }
    override fun getResponseType(): Any? = SearchFiles.responseType
}

open class HelloResponse
{
    var result:String? = null
}

open class SearchFilesResponse
{
    var results:ArrayList<String> = ArrayList<String>()
}

File: JetbrainsComposeDesktopExample/wwwroot/js/dtos.mjs
/* Options:
Date: 2024-03-01 15:11:58
Version: 8.12
Tip: To override a DTO option, remove "//" prefix before updating
BaseUrl: https://localhost:5001

//AddServiceStackTypes: True
//AddDocAnnotations: True
//AddDescriptionAsComments: True
//IncludeTypes: 
//ExcludeTypes: 
//DefaultImports: 
*/

"use strict";
export class HelloResponse {
    /** @param {{result?:string}} [init] */
    constructor(init) { Object.assign(this, init) }
    /** @type {string} */
    result;
}
export class SearchFilesResponse {
    /** @param {{results?:string[]}} [init] */
    constructor(init) { Object.assign(this, init) }
    /** @type {string[]} */
    results;
}
export class Hello {
    /** @param {{name?:string}} [init] */
    constructor(init) { Object.assign(this, init) }
    /** @type {string} */
    name;
    getTypeName() { return 'Hello' }
    getMethod() { return 'GET' }
    createResponse() { return new HelloResponse() }
}
export class SearchFiles {
    /** @param {{pattern?:string}} [init] */
    constructor(init) { Object.assign(this, init) }
    /** @type {string} */
    pattern;
    getTypeName() { return 'SearchFiles' }
    getMethod() { return 'GET' }
    createResponse() { return new SearchFilesResponse() }
}

